### PR TITLE
Feature/camera proto link

### PR DESCRIPTION
**Pull request by owal7428**
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

Integrated rendering corruption functions into main image generation pipeline. Don't merge until after feature/server-build.

## Verification

Expected images are generated for given test files.

### CODE REVIEW

File: Source/cielim/Private/SimulationDataSourceActor.cpp
Comment (thibaudteil) [inline Source/cielim/Private/SimulationDataSourceActor.cpp]: Could you add a if/else in case the exposure time is None?

File: Source/cielim/Private/CaptureManager.cpp
Comment (thibaudteil) [inline Source/cielim/Private/CaptureManager.cpp]: we'll want to add some protection for None values in these

File: Source/cielim/Private/SimulationDataSourceActor.cpp
Comment (thibaudteil) [inline Source/cielim/Private/SimulationDataSourceActor.cpp:192]: I think this could be the best place to check if not None, and have default values otherwise.
Essentially, no blur, no noise, systemGain of 1 and no cosmic Rays would be the defaults

**Review by thibaudteil — APPROVED** [review]
Small protections to add but looks good otherwise

File: Source/test/test_print_proto.py
Comment (patkenneally) [inline Source/test/test_print_proto.py:1]: :large_orange_diamond: The message on this commit is too long. You can see this when viewing in the PR overview page. Can you consider shortening it?

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (patkenneally) [inline Source/cielim/Private/RenderingFunctionsLibrary.cpp:219]: :large_orange_diamond: Is there a reason for preferring the `uint8` type over the `uint8_t` type? I've normally chosen the later because it guarantees type widths across different platforms. I've likely forgotten this in places and done the same.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nicely done. Just a few questions.

File: Source/cielim/Private/RenderingFunctionsLibrary.cpp
Comment (owal7428) [reply Source/cielim/Private/RenderingFunctionsLibrary.cpp:219]: I believe that uint8 is a type specifically for use in Unreal, so I am using that. They should be effectively equivalent.

File: Source/test/test_print_proto.py
Comment (owal7428) [reply Source/test/test_print_proto.py:1]: I've shortened the commit message.

File: Source/cielim/Private/SimulationDataSourceActor.cpp
Comment (owal7428) [reply Source/cielim/Private/SimulationDataSourceActor.cpp]: I've added a ternary statement that sets the lux to 1280 if exposure time is 0 (default value if not defined in the protobuf).

File: Source/cielim/Private/CaptureManager.cpp
Comment (owal7428) [reply Source/cielim/Private/CaptureManager.cpp]: I've added protections in the render functions.

File: Source/cielim/Private/SimulationDataSourceActor.cpp
Comment (owal7428) [reply Source/cielim/Private/SimulationDataSourceActor.cpp:192]: This has been done in the render functions to avoid cluttering the getPNG function.

**Pull request merged**
